src/pages/workspaces/workspace/Data.js

```diff
@@ -75,6 +75,10 @@ const applyColumnSettings = (columnSettings, columns) => {
   )(columns)
 }
 
+const saveScroll = (initialX, initialY) => {
+  StateHistory.update({ initialX, initialY })
```

Comment:
We should be careful about performance here. There is a fair bit of work happening in `update`, including JSON parsing, and this can get called rapidly during scrolling. We might want to `_.debounce` it.